cfme/utils/fileserver.py

```diff
@@ -0,0 +1,190 @@
+        self.entrypoint = entrypoint or ftp_data.entrypoint
+
+        self.client = FTPClient(self.host, self.user, self.passwd, time_diff=False)
+        self.cwd = os.path.join(self.entrypoint, "miq", self.entity if entity else "")
+        self.client.cwd(self.cwd)
+
+    @property
+    def _entities(self):
+        result = {"files": [], "dirs": []}
+
+        def _checks(line):
+            entity = re.split(r"\s+", line, maxsplit=8)[-1]
+            if line.startswith("d"):
+                result["dirs"].append(entity)
+            else:
+                result["files"].append(entity)
+
+        self.client.ftp.dir(_checks)
+        return result
+
+    @property
+    def files(self):
+        """List of FTP file objects"""
+        return [File(name=f, client=self) for f in self._entities["files"]]
+
+    def get_file(self, name):
+        """
+        Arg:
+            name: name of remote file
+        Returns:
+            FTP file object
+        """
+
+        if name in self._entities["files"]:
+            return File(name=name, client=self)
+        else:
+            raise FTPException("{} not found".format(name))
+
+    def rm(self, name):
+        """ Remove or Delete remote FTP file
+        Arg:
+            name: name of remote file
+        Returns:
+            Success of the action
+        """
+        return self.client.dele(name)
+
+    def upload(self, path, name=None):
+        """ Upload FTP file
+        Arg:
+            path: path of local file
+            name: set name of file default it will consider original name of uploading file
+        Returns:
+            Success of the action
+        """
+
+        name = name or os.path.basename(path)
+        if name in self._entities["files"]:
+            raise FTPException("{} already available in {}".format(name, self.entity))
+        with open(path, "rb") as f:
+            return self.client.storbinary(name, f)
+
+    def download(self, name, target=None):
+        """ Download FTP file
+        Arg:
+            name: remote file name
+            target: local path for download else it will consider current working path
+        Returns:
+            target path
+        """
+
+        target = target if target else os.path.join("/tmp", name)
+        if name not in self._entities["files"]:
+            raise FTPException("{} not found in {}".format(name, self.entity))
+
+        with open(target, "wb") as output:
+            self.client.retrbinary(name, output.write)
+            return target
+
+    @property
+    def directories(self):
+        """ List remote FTP directories"""
+        return self._entities["dirs"]
+
+    def mkdir(self, name):
+        """ Create remote FTP directory
+        Arg:
+            name: name of remote directory
+        Returns:
+            Success of the action
+        """
+        return self.client.mkd(name)
+
+    def rmd(self, name):
```

Comment:
How is this different from def rm () at line 97 ? 
We can use only one command and have one method ?